master/buildbot/db/dbconfig.py

```diff
@@ -61,7 +61,8 @@ def getDb(self):
         db.model = model.Model(db)
         db.state = state.StateConnectorComponent(db)
         try:
-            self.objectid = db.state.thdGetObjectId(db_engine, self.name, "DbConfig")['id']
+            with db_engine.connect() as conn:
```

Comment:
The thread pool is to ensure that the main thread does not block. Is db_engine.connect() guaranteed not to block even for a short time?

Reply:
I think this does not change the blocking aspect of this code as `thdGetObjectId` is the "implementation` with `_getObjectId` being the wrapper that will thread it.

I though direct call to `thdGetObjectId` would block the thread as well, am I wrong?

Reply:
OK, I was confused about which code path this is used in. All good, thanks.